### PR TITLE
Split of doc_fragments for virt options

### PR DESCRIPTION
**Pull request by JensHeinrich**
##### SUMMARY

This moves the options for the virt.py module into doc_fragments and reuses them for virt_net and virt_pool.
This is to facilitate reuse and further developments of the modules, eg to split into per command modules (+_info  modules)
 

##### ISSUE TYPE

- Docs Pull Request
##### COMPONENT NAME

virt.py

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Comment by odyssey4me** [comment]
This is great work, thank you - could you rebase this and ensure that the sanity tests pass?

**Comment by JensHeinrich** [comment]
> This is great work, thank you - could you rebase this and ensure that the sanity tests pass?

Yep, sure.
Give me a day or so

**Comment by JensHeinrich** [comment]
Though for the sanity checks I will have to see @odyssey4me as they seem to be partly failing on the main branch too

**Comment by odyssey4me** [comment]
OK, I'll see if I can spend some time this w/end to figure out what's going on and get those fixed.

**Comment by Andersson007** [comment]
recheck

**Comment by Andersson007** [comment]
i closed-reopened it to re-run the tests

**Comment by Andersson007** [comment]
@JensHeinrich hi, thanks for your work! The tests should be fixed now. Could you please rebase your PR https://docs.ansible.com/ansible/latest/dev_guide/developing_rebasing.html . Thank you

**Comment by JensHeinrich** [comment]
> @JensHeinrich hi, thanks for your work! The tests should be fixed now. Could you please rebase your PR https://docs.ansible.com/ansible/latest/dev_guide/developing_rebasing.html . Thank you

Done

**Comment by Andersson007** [comment]
@JensHeinrich thanks! There are couple of sanity issues to fix like:
```
02:16 ERROR: Found 2 pep8 issue(s) which need to be resolved:
02:16 ERROR: plugins/doc_fragments/virt.py:7:1: E302: expected 2 blank lines, found 1 (100%)
02:16 ERROR: plugins/doc_fragments/virt.py:41:1: W293: blank line contains whitespace (100%)
02:27 ERROR: Found 1 pylint issue(s) which need to be resolved:
02:27 ERROR: plugins/doc_fragments/virt.py:41:0: trailing-whitespace: Trailing whitespace (100%)
```
Could you please take a look? (for details click on "Details" in the test box above)

**Comment by Andersson007** [comment]
@JensHeinrich thanks! Now they are green

@odyssey4me as i can see in the first comment the changes work for you, right?

We'll wait until tomorrow and merge

**Comment by Andersson007** [comment]
I've just got feedback from @bcoca 
```
its better as separate files
3:40 PM 
the .var functionality was not planned and has broken in past
```
@JensHeinrich , is it possible to rearrange the work to use separate files?

**Comment by Andersson007** [comment]
@JensHeinrich if it's hard or doesn't make sense, feel free to leave it as it is. Let's wait until tomorrow and merge if there are no objections. Thanks

**Comment by JensHeinrich** [comment]
> @JensHeinrich if it's hard or doesn't make sense, feel free to leave it as it is. Let's wait until tomorrow and merge if there are no objections. Thanks

Wait with the merge, I will look at it it tomorrow morning

**Comment by Andersson007** [comment]
@JensHeinrich sure, thanks! When it's ready, please let me know

**Comment by JensHeinrich** [comment]
> I've just got feedback from @bcoca
> 
> ```
> its better as separate files
> 3:40 PM 
> the .var functionality was not planned and has broken in past
> ```
> 

Is this meant as in regards to importing from multiple files or the vars section in the docs?

**Comment by Andersson007** [comment]
It was said about this particular PR in its original variant.

Also the related discussion https://github.com/ansible/ansible/issues/75285

It should be fine now, I think. Ping me when you finish

**Comment by Andersson007** [comment]
I'm getting the following error when running:
```
$ ansible-doc libvirt_qemu -M ~/ansible_collections/community/libvirt/plugins/connection/
ERROR! module libvirt_qemu missing documentation (or could not parse documentation): unknown doc_fragment(s) in file /home/aklychko/ansible_collections/community/libvirt/plugins/connection/libvirt_qemu.py: community.libvirt.virt_remote_addr, community.libvirt.virt_uri, community.libvirt.virt_executable
```

File: plugins/modules/virt.py
Comment (odyssey4me) [inline plugins/modules/virt.py:28]: python >= 2.6 & libvirt-python are common requirements across the board - can this be moved into a doc fragment too? TBH the python version requirement is something that needs validating - I'm not sure any of that is tested.

File: plugins/modules/virt_net.py
Comment (odyssey4me) [inline plugins/modules/virt_net.py:57]: This is wrong, it should be libvirt-python

File: plugins/modules/virt_pool.py
Comment (odyssey4me) [inline plugins/modules/virt_pool.py:63]: This is wrong, it should be libvirt-python

File: plugins/modules/virt.py
Comment (Andersson007) [reply plugins/modules/virt.py:28]: sounds very sensible to me

Comment (JensHeinrich) [reply plugins/modules/virt.py:28]: Though the other modules also list `python-lxml` here

Comment (abadger) [reply plugins/modules/virt.py:28]: Note that if the python requirement is >= 2.6, it matches ansible's minimum version for managed side code (ie: modules) so it can be omitted.

**Comment by Andersson007** [comment]
@JensHeinrich I think you can revert your last changes and use the original approach you suggested. Sorry

File: plugins/modules/virt_pool.py
Comment (Andersson007) [reply plugins/modules/virt_pool.py:63]: we could fix it in a separate PR, thanks for pointing this out

File: plugins/modules/virt_net.py
Comment (Andersson007) [reply plugins/modules/virt_net.py:57]: we could fix it in a separate PR, thanks for pointing this out

**Comment by Andersson007** [comment]
@JensHeinrich are we waiting for something? I think the original approach is widely used as we can see in the issue, so feel free to revert the last commits to the original state and we can merge the PR

File: plugins/modules/virt_net.py
Comment (JensHeinrich) [reply plugins/modules/virt_net.py:57]: I think thats a good idea to make another PR for that

**Comment by Andersson007** [comment]
FYI: we merged a PR that fixes the doc yesterday. That's why we have conflicts now.

**Comment by JensHeinrich** [comment]
> FYI: we merged a PR that fixes the doc yesterday. That's why we have conflicts now.

Thanks just rebased

**Comment by JensHeinrich** [comment]
rebased on the wrong remote...

**Comment by Andersson007** [comment]
@JensHeinrich if it's hard to fix it now, feel free to close this one and create a new one instead.

**Comment by JensHeinrich** [comment]
It was just doing a side by side of the doc fragment vs the merge conflict

**Comment by Andersson007** [comment]
how can i test this locally?
I get errors, when I run:
```
$ ansible-doc virt -M plugins/modules/
ERROR! module virt missing documentation (or could not parse documentation): unknown doc_fragment(s) in file /home/aklychko/ansible_collections/community/libvirt/plugins/modules/virt.py: community.libvirt.virt.options_uri, community.libvirt.virt.options_xml, community.libvirt.virt.options_guest, community.libvirt.virt.options_autostart, community.libvirt.virt.options_state, community.libvirt.virt.options_command
```

**Comment by Andersson007** [comment]
Thanks tadeboro:
```
You will need to set ANSIBLE_COLLECTIONS_PATH env var and drop the -M flag.
9:56 AM 
`ANSIBLE_COLLECTIONS_PATH=/home/aklychko ansible-doc community.dns.hetzner_dns_record` should work.
```

I'll check all the modules and then will merge

**Review by Andersson007 — APPROVED** [review]

**Comment by Andersson007** [comment]
@JensHeinrich thanks for the contribution! Would you like to continue and fix what @odyssey4me spotted? The doc improvements / fixes are much appreciated!

@odyssey4me @abadger thanks for reviewing!

**Comment by JensHeinrich** [comment]
Can do, but don't know when ATM

**Comment by Andersson007** [comment]
Cool, thanks!

**Comment by Andersson007** [comment]
@JensHeinrich , created an issue https://github.com/ansible-collections/community.libvirt/issues/86 not to forget

**Comment by Andersson007** [comment]
please put your intention in the issue as well to avoid working in parallel with someone else, thanks

**Pull request merged**
